### PR TITLE
Users/devika/addingclass for debuging node

### PR DESCRIPTION
**Pull request by Devika-Tantry**
1. Added placeholder class for debugPoint currently it takes care of this new instruction  `DebugPoint(String)` and just return next instruction. Added simple hello world test for the same.
2. Added the parameter `isDebugEnabled` to `LoadViA` and subsequent method till `ParseClumps` . This parameter is used to ignore the `DebugPoint` instruction if it is set to false. By default, it is set to false.  
3. Changes to release build to emit source map using `-g` will help debug the feature branch.

### CODE REVIEW

File: source/core/CEntryPoints.cpp [hunk 1]
```diff
@@ -28,13 +28,13 @@ VIREO_EXPORT void* EggShell_Create(TypeManagerRef parent)
     return TypeManager::New(parent);
 }
 //------------------------------------------------------------
-VIREO_EXPORT NIError EggShell_REPL(TypeManagerRef tm, const Utf8Char* commands, Int32 length)
+VIREO_EXPORT NIError EggShell_REPL(TypeManagerRef tm, const Utf8Char* commands, Int32 length, Boolean IsDebugEnabled)
```
Comment (shivaprasad-basavaraj) [inline source/core/CEntryPoints.cpp]: [nit] New parameter should be isDebuggingEnabled

File: source/core/TDCodecVia.cpp [hunk 2]
```diff
@@ -1592,7 +1592,7 @@ void TDViaParser::ParseVirtualInstrument(TypeRef viType, void* pData)
     // The clumps code will be loaded once the module is finalized.
 }
 //------------------------------------------------------------
-void TDViaParser::FinalizeVILoad(VirtualInstrument* vi, EventLog* pLog)
+void TDViaParser::FinalizeVILoad(VirtualInstrument* vi, EventLog* pLog, Boolean idIDE)
```
Comment (shivaprasad-basavaraj) [inline source/core/TDCodecVia.cpp]: Vireo code shouldn't have terminology such as IDE. Name this as isDebuggingEnabled as well.

File: source/include/TDCodecVia.h [hunk 3]
```diff
@@ -118,22 +118,22 @@ class TDViaParser
     TypeRef ParseLiteral(TypeRef patternType);
     Int32   ParseData(TypeRef type, void* pData);
     Boolean EatJSONPath(SubString* path);
-    NIError ParseREPL();
+    NIError ParseREPL(Boolean IsDebugEnabled = false);
```
Comment (shivaprasad-basavaraj) [inline source/include/TDCodecVia.h]: All these places should be in camel case since they arguments to the function.

**Review by shivaprasad-basavaraj — APPROVED** [review]

File: source/core/CEntryPoints.cpp (hunk 1 above)
Comment (Devika-Tantry) [reply source/core/CEntryPoints.cpp]: Done

**Pull request merged**